File: dotnet/api-okta-demo/src/Controller/ProfileController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace api_okta_demo.Controller;

[ApiController]
[Route("/api/profile")]
public class ProfileController : ControllerBase
{
    // Public — anyone can call this
    [HttpGet("public")]
    public IActionResult Public()
        => Ok(new { message = "This is public!" });

    // Protected — requires a valid Okta JWT
    [HttpGet("me")]
    [Authorize]
    public IActionResult GetMe()
    {
        // User.Identity is populated from the JWT claims
        var name  = User.FindFirst("name")?.Value;
        var email = User.FindFirst("userEmail")?.Value;
        
        var allClaims = User.Claims.Select(c => new { c.Type, c.Value });

        return Ok(new {
            message = "You are authenticated!",
            name,
            email
        });
    }
}

File: spring-boot/api-okta-demo-service/src/main/java/com/api_okta_demo_service/controller/ProfileController.java
package com.api_okta_demo_service.controller;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.oauth2.jwt.Jwt;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/api/profile")
public class ProfileController {

    @GetMapping("/public")
    public Map<String, String> publicEndpoint() {
        return Map.of("message", "This is public");
    }

    @GetMapping("/me")
    public Map<String, String> getMe(@AuthenticationPrincipal Jwt jwt) {
        String name = jwt.getClaimAsString("name");
        String userEmail = jwt.getClaimAsString("userEmail");

        return Map.of(
                "message",   "You are authenticated!",
                "name",      name != null ? name : "",
                "email", userEmail != null ? userEmail : ""
        );
    }
}
